Sources/Basic/Process.swift

```diff
@@ -348,7 +348,7 @@ public final class Process: ObjectIdentifierProtocol {
             posix_spawn_file_actions_adddup2(&fileActions, 2, 2)
         }
 
-        let argv = CStringArray(arguments)
+        let argv = CStringArray([executablePath.asString] + arguments.dropFirst())
```

Comment:
Weill the test not work without this change? I can't recall if there was a reason to this

Reply:
Without this, we won't actually be using the `PATH` override and instead just execute `pkg-config` so we would again be relying on `PATH` in the process environment.

Reply:
I guess saying "process" here is ambiguous. I meant the environment of the package manager process which we aren't mutating, not the environment of the `Process` instance.

Reply:
I'll take a closer look at the PR tomm., I want to understand whats going on here.

Reply:
I guess my question is why isn't posix_spawn looking up the executable in PATH without change since we call Process with the overridden env?

Reply:
The executable will be looked up using the current processes environment, not the one we pass to `posix_spawn` because that is supposed to be the environment for the new process only.